app/main.py

```diff
@@ -55,13 +55,7 @@ def __eq__(self, other: Distance | int | float) -> bool:
             return self.km == other.km
 
     def __le__(self, other: Distance | int | float) -> bool:
-        if isinstance(other, (int, float)):
-            return self.km <= other
-        elif isinstance(other, Distance):
-            return self.km <= other.km
+        return not self.__gt__(other)
```

Comment:
it's not a good way to call magic methods, you don't add numbers using __add__, right?
it's better to use `return not self > other`